node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -478,82 +483,268 @@ mod tests {
+                                "Invalid initial timer scheduled for {:?}.",
+                                vidx
+                            )
+                        }
+                        timers.insert((timestamp, vidx));
+                    } else {
+                        panic!("Only one initial effect expected")
+                    }
+                    (vidx, av)
+                })
+                .collect();
+
+            TestState {
+                rng,
+                state,
+                instance_id,
+                fd,
+                validators: validator_map,
+                timers,
+            }
+        }
+
+        /// Force the validator to handle timer that may not have been sheduled by it.
+        /// Useful for testing.
+        /// Returns effects created when handling the timer.
+        fn handle_timer(
+            &mut self,
+            validator: ValidatorIndex,
+            timestamp: Timestamp,
+        ) -> Vec<Effect<TestContext>> {
+            // Remove the timer from the queue if it has been scheduled.
+            let _ = self.timers.remove(&(timestamp, validator));
+            let effects = self.validators.get_mut(&validator).unwrap().handle_timer(
+                timestamp,
+                &self.state,
+                self.instance_id,
+                &mut self.rng,
+            );
+            self.schedule_timer(validator, &effects);
+            self.add_new_unit(&effects);
+            effects
+        }
+
+        /// Propose new consensus value as validator `vidx`.
+        /// Returns effects created when proposing and newly proposed wire unit.
+        fn propose(
+            &mut self,
+            vidx: ValidatorIndex,
+            cv: <TestContext as Context>::ConsensusValue,
+            block_context: BlockContext,
+        ) -> (Vec<Effect<TestContext>>, SignedWireUnit<TestContext>) {
+            let validator = self.validators.get_mut(&vidx).unwrap();
+            let proposal_timestamp = block_context.timestamp();
+            let effects = validator.propose(
+                cv,
+                block_context,
+                &self.state,
+                self.instance_id,
+                &mut self.rng,
+            );
+
+            // Add the new unit to the state.
+            let proposal_wunit = unwrap_single(&effects).unwrap_unit();
+            let prop_hash = proposal_wunit.hash();
+            self.state.add_unit(proposal_wunit.clone()).unwrap();
+            let effects = validator.on_new_unit(
+                &prop_hash,
+                proposal_timestamp + 1.into(),
+                &self.state,
+                self.instance_id,
+                &mut self.rng,
+            );
+            self.schedule_timer(vidx, &effects);
+            (effects, proposal_wunit)
+        }
+
+        /// Handle new unit by validator `vidx`.
+        /// Since all validator use the same state, that unit should be added already. Panics if
```

Comment:
```suggestion
        /// Since all validators use the same state, that unit should be added already. Panics if
```

Reply:
Fixed in [`3e59e5b` (#602)](https://github.com/CasperLabs/casper-node/pull/602/commits/3e59e5be187011e0da8f7801704d291ba64967fc).